### PR TITLE
Optimize jni calling overhead

### PR DESCRIPTION
**Pull request by ted-wq-x**
I read and agree to the terms under [CLA.md](https://github.com/kuzudb/kuzu/blob/master/CLA.md)

In my test environment ,QPS from 250 to 750

### CODE REVIEW

File: tools/java_api/src/jni/kuzu_java.cpp [hunk 1]
```diff
@@ -1265,3 +1225,246 @@ JNIEXPORT jstring JNICALL Java_com_kuzudb_KuzuNative_kuzu_1get_1version(JNIEnv*
...
+    env->DeleteGlobalRef(J_C_Short);
+    env->DeleteGlobalRef(J_C_Byte);
+    env->DeleteGlobalRef(J_C_BigInteger);
+    env->DeleteGlobalRef(J_C_Float);
+    env->DeleteGlobalRef(J_C_Duration);
+    env->DeleteGlobalRef(J_C_UUID);
+    env->DeleteGlobalRef(J_C_KuzuConnection);
+    env->DeleteGlobalRef(J_C_KuzuDatabase);
+    env->DeleteGlobalRef(J_C_String);
+}
```
Comment (mewim) [inline tools/java_api/src/jni/kuzu_java.cpp:1470]: Missing a newline here

Comment (ted-wq-x) [reply tools/java_api/src/jni/kuzu_java.cpp]: done

File: tools/java_api/src/jni/kuzu_java.cpp [hunk 2]
```diff
@@ -1265,3 +1225,246 @@ JNIEXPORT jstring JNICALL Java_com_kuzudb_KuzuNative_kuzu_1get_1version(JNIEnv*
 JNIEXPORT jlong JNICALL Java_com_kuzudb_KuzuNative_kuzu_1get_1storage_1version(JNIEnv*, jclass) {
     return static_cast<jlong>(Version::getStorageVersion());
 }
+
+void initGlobalClassRef(JNIEnv* env) {
+    jclass tempLocalClassRef;
+    tempLocalClassRef = env->FindClass("java/util/Map");
+    J_C_Map = (jclass)env->NewGlobalRef(tempLocalClassRef);
+    env->DeleteLocalRef(tempLocalClassRef);
```
[the remark this answers is not in the document]
Comment (ted-wq-x) [reply tools/java_api/src/jni/kuzu_java.cpp]: done

**Review by mewim — APPROVED** [review]
OK. This LGTM now and the tests are passing. I think it's ready to be merged.

**Pull request merged**